trace-obfuscation/src/credit_cards.rs

```diff
@@ -0,0 +1,422 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+/// is_card_number checks if b could be a credit card number by checking the digit count and IIN prefix.
+/// If validateLuhn is true, the Luhn checksum is also applied to potential candidates.
+/// Note: This code is based on the code from datadog-agent/pkg/obfuscate/credit_cards.go
+pub fn is_card_number(s: &str, validate_luhn: bool) -> bool {
+    if s.len() < 12 {
+        // fast path
+        return false;
+    }
+
+    // Only valid characters are 0-9, space (" ") and dash("-")
+    let num_s = s.replace([' ', '-'], "");
```

Comment:
since its a hackathon, I'll dare to propose, doing one single loop. 
something like:
```rust 
let mut num_s = Vec::with_capacity(s.len());
for i in 0..s.len() {
let c = s[i];
match c {
  ' ' | '-' =>  continue
  _ => {}
}
if !c.is_numeric {
return false
}
num_s.push(c);
}

Reply:
Doing this in a single loop was so much better, it also meant I could move the `to_digit` call from two places to a single place and then simplified my arguments so I didn't have to move and copy things around quite so much.